marshal/consumer.go

```diff
@@ -318,6 +352,17 @@ func (c *Consumer) claimTopics() {
 		} else {
 			// Unclaimed, so attempt to claim partition 0. This is how we key topic claims.
 			log.Infof("[%s] attempting to claim topic (key partition 0)\n", topic)
+			// we need to check if we're above the maximum topics to be claimed
+			// we should only allow the first k topics to be claimed and allow all
+			// of their partitions to be claimed. This is controlled by controlling how
+			// many (key partition 0) we claim.
+
+			if c.isTopicClaimReached(topic) {
+				log.Debugf("blocked claiming topic: %s due to limit %d\n",
```

Comment:
I didn't want to pollute the logs since this will happen every 3 seconds....that's why I put it in Debug...will switch it back